File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.py
def solution(n):
    answer = []
    
    def hanoi(count, start, temp, to):
        if(count == 1):
            answer.append([start, to])
            return
        hanoi(count-1, start, to, temp)
        answer.append([start, to])
        hanoi(count-1, temp, start, to)
    
    hanoi(n, 1, 2, 3)
    return answer

File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.js
function solution(n) {
    var answer = [];
    
    
    function hanoi(count, from, temp, to){
        if(count === 1) {
            answer.push([from, to]);
            return;
        }
        hanoi(count-1, from, to, temp);
        answer.push([from, to]);
        hanoi(count-1, temp, from, to);
    }
    
    hanoi(n, 1, 2, 3);
    return answer;
}